### PR TITLE
Calcite Joins regression tests. WIP

### PR DESCRIPTION
**Pull request by alexeevm**

### CODE REVIEW

**Comment by kaikaizi** [comment]
Related: I made this PR: https://github.com/VoltDB/voltdb/pull/6607/files to clean up some parts related to merge join executor.

**Comment by kaikaizi** [comment]
[My patch to apply](https://github.com/VoltDB/voltdb/files/3664288/0001-patch-to-apply-to-successfully-merge-from-latest-mas.patch.txt). Should work with or wo. merging from latest master branch, either way.

File: src/frontend/org/voltdb/plannerv2/rules/PlannerRules.java [hunk 1]
```diff
@@ -209,12 +209,12 @@ public RuleSet getRules() {
             MPSetOpsQueryFallBackRule.INSTANCE
     );
 
-    private static final RuleSet HEP_OUTER_JOIN = RuleSets.ofList(
+    private static final RuleSet HEP_LOGICAL_JOIN = RuleSets.ofList(
             CalcMergeRule.INSTANCE,
-            VoltLJoinCommuteRule.INSTANCE_RIGHT_TO_LEFT
+            VoltLJoinCommuteRule.INSTANCE_RIGHT_TO_LEFT,
+            VoltLCalcJoinMergeRule.INSTANCE
```
Comment (kaikaizi) [inline src/frontend/org/voltdb/plannerv2/rules/PlannerRules.java]: Did you forget to git-add this file in commit "Handle WHERE join conditions for MJ and NLJ"?

Comment (alexeevm) [reply src/frontend/org/voltdb/plannerv2/rules/PlannerRules.java]: Yes, thank you. Just added the missing VoltLCalcJoinMergeRule.java

**Comment by kaikaizi** [comment]
Merged in #6686

**Pull request closed without merging**